### PR TITLE
Fix potential deadlock when HttpPageBufferClient exceeds memory limit

### PR DESCRIPTION
**Pull request by pettyjamesm**
Comparable changes to https://github.com/prestodb/presto/pull/15220

Prior to this change, if the HttpPageBufferClient caused a task to exceed its memory limit inside of addPages, the exception would bubble to the root of the exchange client callback executor without ever marking the task as failed. If no other operator also saw the memory limit exceeded condition, then the task would become stuck in a deadlocked
state.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java [hunk 1]
```diff
@@ -387,26 +387,26 @@ public Void handle(Request request, Response response)
                             }
                         });
                     }
+
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java:390]: Could we have a test in `io.prestosql.operator.TestHttpPageBufferClient` for this case (e.g the exception did not bubble up)?

Comment (pettyjamesm) [reply presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java:390]: Test added

File: presto-main/src/test/java/io/prestosql/operator/TestHttpPageBufferClient.java [hunk 2]
```diff
@@ -237,7 +244,7 @@ public void testInvalidResponses()
                 "localhost",
                 new TestingHttpClient(processor, scheduler),
                 DataIntegrityVerification.ABORT,
-                DataSize.of(10, Unit.MEGABYTE),
```
Comment (sopel39) [inline presto-main/src/test/java/io/prestosql/operator/TestHttpPageBufferClient.java:240]: nit: this could be separate commit

**Review by sopel39 — APPROVED** [review]

**Comment by sopel39** [comment]
merged, thanks!
